databroker/tests/test_v2/generic.py

```diff
@@ -28,26 +28,46 @@ def compare(a, b):
     a_indexed = {}
     b_indexed = {}
     for name, doc in a:
-        if name == 'datum':
+        if name == 'resource':
+            # Check for an extraneous duplicate key in old documents.
+            if 'id' in doc:
+                assert doc['id'] == doc['uid']
+                doc = doc.copy()
+                doc.pop('id')
```

Comment:
The resource isn't being added to a_indexed, same below.

Reply:
See https://github.com/bluesky/databroker/pull/519#discussion_r380917557